### PR TITLE
add back buttons and prepare ShuftiPro

### PR DESCRIPTION
**Pull request by jpangelle**
### Description

This pull request integrates the ability to view previous steps. Also prepares the app for the ShuftiPro integration.

### Approvals

- [x] Dev

### CODE REVIEW

File: centrifuge-app/src/pages/Onboarding/BusinessInformation.tsx [hunk 1]
```diff
@@ -41,17 +54,29 @@ const BusinessInformationInlineFeedback = ({ isError }: { isError: boolean }) =>
...
 }
 
-export const BusinessInformation = ({ nextStep, setUltimateBeneficialOwners }: Props) => {
+export const BusinessInformation = ({ backStep, nextStep }: Props) => {
   const { selectedAccount } = useWeb3()
   const { authToken } = useAuth()
+  const { onboardingUser, refetchOnboardingUser } = useOnboardingUser() as {
+    onboardingUser: EntityUser
+    refetchOnboardingUser: () => void
+  }
```
Comment (sophialittlejohn) [inline centrifuge-app/src/pages/Onboarding/BusinessInformation.tsx:63]: I think using a generic type for the hook would be nicer than type casting it here

File: centrifuge-app/src/pages/Onboarding/BusinessInformation.tsx [hunk 2]
```diff
@@ -41,17 +54,29 @@ const BusinessInformationInlineFeedback = ({ isError }: { isError: boolean }) =>
...
+
+  const isCompleted = useMemo(() => {
+    if (onboardingUser?.steps?.verifyBusiness.completed) {
+      return true
+    }
+
+    return false
+  }, [onboardingUser])
```
Comment (sophialittlejohn) [inline centrifuge-app/src/pages/Onboarding/BusinessInformation.tsx:71]:
```suggestion
  const isCompleted = !!onboardingUser?.steps?.verifyBusiness.completed
```
Don't think useMemo is really necessary here

File: centrifuge-app/src/pages/Onboarding/BusinessOwnership.tsx [hunk 3]
```diff
@@ -52,23 +57,37 @@ const BusinessOwnershipInlineFeedback = ({ isError }: { isError: boolean }) => {
...
+    refetchOnboardingUser: () => void
+  }
+
+  const isCompleted = useMemo(() => {
+    if (onboardingUser?.steps?.confirmOwners.completed) {
+      return true
+    }
+
+    return false
+  }, [onboardingUser])
```
Comment (sophialittlejohn) [inline centrifuge-app/src/pages/Onboarding/BusinessOwnership.tsx:74]: same here, don't think memo is necessary

File: centrifuge-app/src/pages/Onboarding/InvestorType.tsx [hunk 4]
```diff
@@ -1,59 +1,86 @@
...
+  const { onboardingUser } = useOnboardingUser()
+
+  const isDisabled = useMemo(() => {
+    if (onboardingUser?.investorType) {
+      return true
     }
-  }, [investorType, nextStep])
+
+    return false
+  }, [onboardingUser])
```
Comment (sophialittlejohn) [inline centrifuge-app/src/pages/Onboarding/InvestorType.tsx:54]: also here 😄

**Review by sophialittlejohn — COMMENTED** [review]
Nice!! 😍

File: centrifuge-app/src/pages/Onboarding/BusinessInformation.tsx (hunk 2 above)
Comment (jpangelle) [reply centrifuge-app/src/pages/Onboarding/BusinessInformation.tsx:71]: 👍

File: centrifuge-app/src/components/OnboardingUserProvider.tsx [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+const AUTHORIZED_ONBOARDING_PROXY_TYPES = ['Any', 'Invest', 'NonTransfer', 'NonProxy']
+
+const OnboardingUserContext = React.createContext<{
+  onboardingUser: OnboardingUser
+  refetchOnboardingUser: () => void
+  isOnboardingUserLoading: boolean
+  isOnboardingUserFetched: boolean
+} | null>(null)
+
+export const OnboardingUserProvider = ({ children }: { children: React.ReactNode }) => {
```
[the remark this answers is not in the document]
Comment (jpangelle) [reply centrifuge-app/src/components/OnboardingUserProvider.tsx:16]: noiceee

**Review by sophialittlejohn — APPROVED** [review]
💯

**Pull request merged**